lib/nerves_runtime/log_tailer.ex

```diff
@@ -0,0 +1,137 @@
+defmodule Nerves.Runtime.LogTailer do
+  use GenServer
+
+  require Logger
+
+  defp gen_server_name(:syslog), do: __MODULE__.Syslog
+  defp gen_server_name(:kmsg), do: __MODULE__.Kmsg
+
+  @doc """
+  `type` must be `:syslog` or `:kmsg` to indicate which log to tail with this
+  process. They're manages by separate processes both to isolate failures and
+  to simplify the handling of messages being sent back from the ports.
+  """
+  @spec start_link(:syslog | :kmsg) :: {:ok, pid()}
+  def start_link(type) do
+    GenServer.start_link(__MODULE__, type, name: gen_server_name(type))
+  end
+
+  @spec init(:syslog | :kmsg) :: {:ok, %{type: atom(), port: port(), buffer: binary()}}
+  def init(type) do
+    if Application.get_env(:nerves_runtime, :disable_syslog, false) do
```

Comment:
Use pattern matching here to delegate two separate branches into their own functions.